kernel/src/process/posix_thread/exit.rs

```diff
@@ -143,6 +143,9 @@ fn wake_robust_list(thread_local: &ThreadLocal, tid: Tid) {
 
     trace!("exit: wake up the rubust list: {:?}", list_head);
     for futex_addr in list_head.futexes() {
+        if futex_addr % align_of::<u32>() != 0 {
```

Comment:
https://github.com/asterinas/asterinas/blob/c4e8bf2c21aa8ad4b272dbc1d68461fa3b42004e/kernel/src/process/posix_thread/robust_list.rs#L132-L135

I think we can fix the code here, by checking something like:
https://github.com/asterinas/asterinas/blob/c4e8bf2c21aa8ad4b272dbc1d68461fa3b42004e/kernel/src/process/posix_thread/futex.rs#L540-L545

---

In addition, I don't think the Linux implementation checks for the zero address, and we don't check it in our `futex()` syscall implementation either. Given these two reasons, perhaps we should simply remove the zero address check? Checking the alignment should be sufficient.